openedx/core/djangoapps/courseware_api/serializers.py

```diff
@@ -56,6 +56,30 @@ class Meta:
         ref_name = 'courseware_api'
 
 
+class CourseProgramSerializer(serializers.Serializer):
+    progress = serializers.SerializerMethodField()
+    title = serializers.CharField()
+    type = serializers.CharField()
+    url = serializers.SerializerMethodField()
+    uuid = serializers.SerializerMethodField()
+
+    def get_progress(self, program):
+        progress = program.get('progress')
+        return {
+            'completed': progress.get('completed'),
+            'in_progress': progress.get('in_progress'),
+            'not_started': progress.get('not_started')
+        }
+
+    def get_url(self, program):
+        request = self.context.get('request')
+        return request.build_absolute_uri(program.get('url'))
+
+    def get_uuid(self, program):
+        progress = program.get('progress')
+        return progress.get('uuid')
```

Comment:
nit: You could also put the uuid at the top level of the context? Feels weird to pull it from the progress block. But super doesn't matter.